src/loaders/JsonImageInfoLoader.ts

```diff
@@ -187,6 +198,35 @@ class JsonImageInfoLoader implements IVolumeLoader {
       const url = imageArray[i].name;
       const batch = imageArray[i].channels;
 
+      // construct cache query
+      const regionPx = convertSubregionToPixels(volume.loadSpec.subregion, volume.imageInfo.subregionSize);
+      const cacheQueryDims = {
+        region: regionPx,
+        time: volume.loadSpec.time,
+        scale: 0,
+      };
+      // Because the data is fetched such that one fetch returns a whole batch,
+      // if any in batch is cached then they all should be. So if any in batch is NOT cached,
+      // then we will have to do a batch request. This logic works both ways because it's all or nothing.
+      let cacheHit = true;
+      for (let j = 0; j < Math.min(batch.length, 4); ++j) {
+        const chindex = batch[j];
+        const cacheResult = cacheStore && cache?.get(cacheStore, chindex, cacheQueryDims);
+        if (cacheResult) {
+          volume.setChannelDataFromVolume(chindex, cacheResult);
+          onChannelLoaded?.(volume, chindex);
+        } else {
+          cacheHit = false;
+          // we can stop checking because we know we are going to have to fetch the whole batch
+          break;
+        }
+      }
+
+      // if there were ANY cache misses then we need to request the batch
+      if (cacheHit) {
+        continue;
+      }
```

Comment:
nit: might flip the phrasing on this comment to match the flipped logic of this `if`, something like "if all channels are in cache, move on to the next image without requesting"